cmd/clusterctl/client/cluster/mover.go

```diff
@@ -1329,7 +1329,13 @@ func applyMutators(object client.Object, mutators ...ResourceMutatorFunc) (*unst
 	}
 	u.SetUnstructuredContent(to)
 	for _, mutator := range mutators {
-		if err := mutator(u); err != nil {
+		var err error
+		if mutator != nil {
+			err = mutator(u)
+		} else {
+			err = fmt.Errorf("mutator is nil")
```

Comment:
```suggestion
			err = errors.New("mutator is nil")
```

Reply:
Fixed in https://github.com/kubernetes-sigs/cluster-api/pull/10951/commits/12b0d07f9962abda86f4549b567c38dd63ab9500